From 9a9523c5cc46655f97f0cf0a713d60c8ed8f30c7 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Mon, 18 Dec 2023 09:58:22 +0100
Subject: [PATCH 0269/1695] drivers: stm32_iwdg: remove stm32_iwdg_refresh()

Remove unused stm32_iwdg_refresh() intended to refresh all registered
watchdog devices.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 0c2a9bd62fe3fdc591a50a99dbab0ee33ab1ca46 drivers: stm32_iwdg: remove stm32_iwdg_refresh()]
Acked-by: Jerome Forissier <jerome.forissier@linaro.org>
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 core/drivers/stm32_iwdg.c         | 14 --------------
 core/include/drivers/stm32_iwdg.h |  4 ----
 2 files changed, 18 deletions(-)

diff --git a/core/drivers/stm32_iwdg.c b/core/drivers/stm32_iwdg.c
index d91882241..43a12b1be 100644
--- a/core/drivers/stm32_iwdg.c
+++ b/core/drivers/stm32_iwdg.c
@@ -90,8 +90,6 @@ struct stm32_iwdg_device {
 	SLIST_ENTRY(stm32_iwdg_device) link;
 };
 
-static unsigned int iwdg_lock = SPINLOCK_UNLOCK;
-
 static SLIST_HEAD(iwdg_dev_list_head, stm32_iwdg_device) iwdg_dev_list =
 	SLIST_HEAD_INITIALIZER(iwdg_dev_list_head);
 
@@ -251,18 +249,6 @@ static const struct wdt_ops stm32_iwdg_ops = {
 };
 DECLARE_KEEP_PAGER(stm32_iwdg_ops);
 
-/* Refresh all registered watchdogs */
-void stm32_iwdg_refresh(void)
-{
-	struct stm32_iwdg_device *iwdg = NULL;
-	uint32_t exceptions = cpu_spin_lock_xsave(&iwdg_lock);
-
-	SLIST_FOREACH(iwdg, &iwdg_dev_list, link)
-		iwdg_refresh(iwdg);
-
-	cpu_spin_unlock_xrestore(&iwdg_lock, exceptions);
-}
-
 /* Driver initialization */
 static TEE_Result stm32_iwdg_parse_fdt(struct stm32_iwdg_device *iwdg,
 				       const void *fdt, int node)
diff --git a/core/include/drivers/stm32_iwdg.h b/core/include/drivers/stm32_iwdg.h
index cab9c9c97..a575a199d 100644
--- a/core/include/drivers/stm32_iwdg.h
+++ b/core/include/drivers/stm32_iwdg.h
@@ -26,8 +26,4 @@ struct stm32_iwdg_otp_data {
  */
 TEE_Result stm32_get_iwdg_otp_config(paddr_t pbase,
 				     struct stm32_iwdg_otp_data *otp_data);
-
-/* Refresh all registered IWDG watchdog instance */
-void stm32_iwdg_refresh(void);
-
 #endif /*__DRIVERS_STM32_IWDG_H*/
-- 
2.47.2

